libshaderc_spvc/src/spvc.cc

```diff
@@ -171,7 +202,88 @@ shaderc_spvc_compilation_result_t validate_and_compile(
 shaderc_spvc_compilation_result_t shaderc_spvc_compile_into_glsl(
     const shaderc_spvc_compiler_t, const uint32_t* source, size_t source_len,
     shaderc_spvc_compile_options_t options) {
+  auto* result = new (std::nothrow) shaderc_spvc_compilation_result;
+  if (!result) return nullptr;
+
   spirv_cross::CompilerGLSL compiler(source, source_len);
+
+  if (options->glsl.version == 0) {
+    // no version requested, was one detected in source?
+    options->glsl.version = compiler.get_common_options().version;
+    if (options->glsl.version == 0) {
+      // no version detected in source, use default
+      options->glsl.version = 450;
+    } else {
+      // version detected implies ES also detected
+      options->glsl.es = compiler.get_common_options().es;
+    }
+  }
+
+  auto entry_points = compiler.get_entry_points_and_stages();
+  spv::ExecutionModel model = spv::ExecutionModelMax;
+  if (!options->entry_point.empty()) {
+    // Make sure there is just one entry point with this name, or the stage is ambiguous.
+    uint32_t stage_count = 0;
+    for (auto &e : entry_points) {
+      if (e.name == options->entry_point) {
+        stage_count++;
+        model = e.execution_model;
+      }
+    }
+    if (stage_count != 1) {
+      result->status = shaderc_compilation_status_compilation_error;
+      if (stage_count)
```

Comment:
Isn't this backwards? This should be if (stage_count == 0)?

Reply:
good catch